tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -229,30 +229,51 @@ void test_variables_construction_and_access() noexcept {
       == "Variables(vector,scalar,scalar2)");
 }
 
-SPECTRE_TEST_CASE("Unit.DataStructures.Variables.Move",
-                  "[DataStructures][Unit]") {
-  Variables<tmpl::list<VariablesTestTags_detail::vector>> x(1, -2.0),
-      z(2, -3.0);
-  CHECK(&get<VariablesTestTags_detail::vector>(z)[0][0] == z.data());
-  Variables<tmpl::list<VariablesTestTags_detail::vector>> y = std::move(x);
-  x = std::move(z);
-  CHECK(
-      (x == Variables<tmpl::list<VariablesTestTags_detail::vector>>{2, -3.0}));
-  CHECK(&get<VariablesTestTags_detail::vector>(x)[0][0] == x.data());
+template <typename VectorType>
+void test_variables_move() noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<
+      typename tt::get_fundamental_type_t<typename VectorType::value_type>>
+      dist{-100, 100};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+
+  size_t number_of_grid_points1 = sdist(gen),
+         number_of_grid_points2 = sdist(gen);
+  std::vector<typename VectorType::value_type> initial_fill_values{2};
```

Comment:
You want a single `std::vector` with the value `2` and then fill it randomly?

Reply:
should be the size with a single value specified. Here I'm trying to aggregate the random generations of similarly used values so that `make_with_random_values` doesn't have to appear so frequently.

Reply:
You are creating a `std::vector` with one element whose value is `2`. Is this what you want or do you want a vector with 2 elements? I think if you use the `make_with_random_value` you can make this const and avoid the confusing mess that is constructing `std::vector`